src/charm.py

```diff
@@ -234,7 +250,7 @@ def _install(self) -> None:
         """Install/refresh the Grafana Agent snap."""
         self.unit.status = MaintenanceStatus("Installing grafana-agent snap")
         try:
-            install_ga_snap(classic=False)
+            install_ga_snap(classic=self.config["classic_snap"])  # type: ignore
```

Comment:
In other charms, we `cast` the config to whatever we need, instead of `type: ignore`. Can we do that here as well?